lib/AST/ASTPrinter.cpp

```diff
@@ -4965,6 +4982,15 @@ void PrintAST::visitDictionaryExpr(DictionaryExpr *expr) {
 }
 
 void PrintAST::visitArrowExpr(ArrowExpr *expr) {
+  visit(expr->getArgsExpr());
+  if (auto isAsync = expr->getAsyncLoc()) {
+    Printer << " async";
+  }
+  if (auto doesThrow = expr->getThrowsLoc()) {
+    Printer << " throws";
+  }
```

Comment:
This works, but I think it would be better spelt as `if (expr->getAsyncLoc().isValid())` & `if (expr->getThrowsLoc().isValid())`.